modules/nf-core/software/samtools/sort/main.nf

```diff
@@ -0,0 +1,35 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process SAMTOOLS_SORT {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::samtools=1.10" : null)
+    if (workflow.containerEngine == 'singularity' && !params.pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/samtools:1.10--h9402c20_2"
+    } else {
+        container "quay.io/biocontainers/samtools:1.10--h9402c20_2"
+    }
+
+    input:
+    tuple val(meta), path(bam)
+    
+    output:
+    tuple val(meta), path("*.bam"), emit: bam

```

Comment:
Can we do CRAM output this way or is this not possible that way?

Reply:
I don't see why not. We could just update the `main.nf` on nf-core/modules to have optional outputs and amend the `script` section accordingly if required. Or the additional arguments to produce CRAM could also come via the `options.args` parameter provided by the developer.